server/api/reviews.go

```diff
@@ -0,0 +1,125 @@
+package handler
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/codeday-labs/bookworms/server/db"
+	"github.com/codeday-labs/bookworms/server/utils"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+type ReviewBody struct {
+	UserName   string   `json:"user_name"`
+	BookName   string   `json:"book_name"`
+	Text       string   `json:"text"`
+	Categories []string `json:"categories"`
+}
+
+// singular collection name from mongodb collections conventions
+const ReviewsCollection = "review"
+
+func filterReviews(filter interface{}) ([]*db.Review, error) {
+
+	// a slice to store decoded reviews
+	var reviews []*db.Review
+
+	cur, err := db.DB().Collection(ReviewsCollection).Find(db.Ctx, filter)
+
+	if err != nil {
+		return reviews, err
+	}
+
+	// Iterate over a returned cursor and decode each at a time
+
+	for cur.Next(db.Ctx) {
+		var r db.Review
+		err := cur.Decode(&r)
+		if err != nil {
+			return reviews, err
+		}
+
+		reviews = append(reviews, &r)
+	}
+
+	if err := cur.Err(); err != nil {
+		return reviews, err
+	}
+
+	// once once done iterating the cursor close
+	cur.Close(db.Ctx)
+
+	if len(reviews) == 0 {
+		return reviews, mongo.ErrNoDocuments
+	}
+
+	return reviews, nil
+}
+
+func getAll() ([]*db.Review, error) {
+	// Passing bson.D{{}} matches all documents in a collection
+	filter := bson.D{{}}
+	return filterReviews(filter)
+}
+
+func createReview(review *db.Review) error {
+	_, err := db.DB().Collection(ReviewsCollection).InsertOne(db.Ctx, review)
+	return err
+}
+
+func Reviews(w http.ResponseWriter, r *http.Request) {
+
+	switch r.Method {
+	case "GET":
+		reviews, err := getAll()
+
+		if err != nil && err != mongo.ErrNoDocuments {
+			utils.RespondWithError(w, "Something went wrong!", http.StatusBadRequest)
+			return
+		}
+
+		db.DB().Client().Disconnect(db.Ctx)
```

Comment:
We might fail to disconnect client here as this function might returned before you reach this line of code.

The recommended way is to move this fucntion to line. 29 with `defer`, so that the connection can be closed right after you finish the task.

```go
cur, err := db.DB().Collection(ReviewsCollection).Find(db.Ctx, filter)
defer db.DB().Client().Disconnect(db.Ctx)
```